### PR TITLE
ensure that query metadata are available cross-folder

### LINKED ISSUE #3463: regression coverage for issue 45664
https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=45664

### PR DESCRIPTION
**Pull request by labkey-chrisj**
#### Rationale
This adds test coverage for[ issue 45664](https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=45664), which resolves the issue where query metadata are not available from subfolder/cross-folder views

#### Related Pull Requests
n/a

#### Changes
just the new test

### CODE REVIEW

File: query/test/src/org/labkey/test/tests/query/CrossFolderDataClassTest.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+    private void doSetup()
+    {
+        _containerHelper.createProject(getProjectName(), null);
+        _containerHelper.createSubfolder(getProjectName(), SUBFOLDER_A);
+        SUBFOLDER_A_PATH = getProjectName() + "/" + SUBFOLDER_A;
+    }
+
+    /**
+     * Issue 45644 addresses the problem where DataClass metadata wasn't available in query when querying cross-folder
```
Comment (labkey-tchad) [inline query/test/src/org/labkey/test/tests/query/CrossFolderDataClassTest.java]: Issue number typo. Also, formatting the comment like "Issue xxxxx:" make TeamCity and IntelliJ link to the issue.
```suggestion
     * Issue 45664: addresses the problem where DataClass metadata wasn't available in query when querying cross-folder
```

File: query/test/src/org/labkey/test/tests/query/CrossFolderDataClassTest.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.*;
+
+/**
+ * use this test as a place to put cross-folder dataclass testing
+ */
+@Category({})
```
Comment (labkey-tchad) [inline query/test/src/org/labkey/test/tests/query/CrossFolderDataClassTest.java]: Add to Daily suite
```suggestion
@Category({Daily.class})
```

**Review by labkey-tchad — CHANGES_REQUESTED** [review]
This doesn't seem to cover the repro steps from the issue.
 - Shouldn't have a container filter
 - Should check for `url` in the name column

Inserting multiple rows seems to be adding unnecessary complication.

File: query/test/src/org/labkey/test/tests/query/CrossFolderDataClassTest.java (hunk 2 above)
Comment (labkey-chrisj) [reply query/test/src/org/labkey/test/tests/query/CrossFolderDataClassTest.java]: done

**Review by labkey-tchad — APPROVED** [review]
Looks good.

**Pull request merged**